File: app/src/main/java/io/github/wulkanowy/ui/splash/SplashPresenter.java
package io.github.wulkanowy.ui.splash;

import android.os.Handler;
import android.support.annotation.NonNull;

import javax.inject.Inject;

import io.github.wulkanowy.data.RepositoryContract;
import io.github.wulkanowy.ui.base.BasePresenter;

public class SplashPresenter extends BasePresenter<SplashContract.View>
        implements SplashContract.Presenter {

    @Inject
    SplashPresenter(RepositoryContract repository) {
        super(repository);
    }

    @Override
    public void onStart(@NonNull SplashContract.View activity) {
        super.onStart(activity);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if (getRepository().getCurrentUserId() == 0) {
                    getView().openLoginActivity();
                } else {
                    getView().startFullSyncService();
                    getView().openMainActivity();
                }
            }
        }, 500);
    }
}

File: app/src/main/java/io/github/wulkanowy/ui/splash/SplashPresenter.kt
package io.github.wulkanowy.ui.splash

import io.github.wulkanowy.data.repositories.StudentRepository
import io.github.wulkanowy.ui.base.BasePresenter
import io.reactivex.disposables.CompositeDisposable
import javax.inject.Inject

class SplashPresenter @Inject constructor(private val studentRepository: StudentRepository,
                                          disposable: CompositeDisposable)
    : BasePresenter<SplashView>(disposable) {

    override fun attachView(view: SplashView) {
        super.attachView(view)
        view.cancelNotifications()

        if (studentRepository.isStudentLoggedIn) {
            view.openMainActivity()
        } else {
            view.openLoginActivity()
        }
    }
}
